_posts/en/newsletters/2019-01-08-newsletter.md

```diff
@@ -0,0 +1,168 @@
+[LND][lnd commits], [C-lightning][cl commits], and [libsecp256k1][secp
+commits].*
+
+- [Bitcoin Core #14565][] significantly improves the error handling for
+  the `importmulti` RPC and will return a `warnings` field for each
+  attempted import with an array of strings describing any problems with
+  the that import (but only if there were any problems).
+
+- [Bitcoin Core #14811][] updates the [getblocktemplate][rpc
+  getblocktemplate] RPC to require that the segwit flag be passed.  This
+  helps prevent miners from accidentally not using segwit, which reduces
+  their fee income.  See [Newsletter #20][] for a recent instance where
+  this may have happened to a large mining pool.
+
+- [Bitcoin Core #14336][] changes the system call (syscall) Bitcoin Core
+  uses to handle network connections and other resources on platforms
+  that support an alternative syscall (only Linux for now).  On a
+  Debian GNU/Linux x86_64 system tested by Optech, this increased the
+  maximum allowed number of connections from 865 to 9,999.  Please note,
+  developers strongly advise against increasing the maximum number of
+  connections unless you have a special need for more connections.  For
+  each peer you have after your first peer, you increase the amount of
+  bandwidth overhead used by your node and by all of your peers.  The
+  default minimum of 8 and maximum of 125 peers were chosen to provide
+  enough connections to ensure robust relaying but also to not create so
+  much bandwidth overhead that the network would be excessively
+  wasteful.  As relay improvements are implemented (see the News section
+  of [Newsletter #26][]), Bitcoin Core developers may adjust these
+  defaults.
+
+- [C-Lightning #2172][] allows `lightningd` to be shutdown normally even
+  if it's operating as the primary process (PID 1), which can be useful
+  in Docker containers.
+
+- [C-Lightning #2188][] adds notification subscription handlers that can
+  be used by plugins, with initial support for notifications that the
+  node has connected to a new peer or disconnected from an existing
+  peer.  The [plugin documentation][cl plugin event] and [sample
+  plugin][cl helloworld.py] have been updated for these handlers.
+
+- [LND #2374][] increases the maximum size of the gRPC messages the
+  `lncli` tool will accept from 4 MB to 50 MB.  This fixes a problem
+  some nodes were encountering where the `describegraph` RPC was failing
+  due to the network having grown so large that messages exceeded this
+  limit.  Developers using gRPC directly will need to increase their
+  client-side maximum message size setting---descriptions of how to do
+  this have already been added as comments to the PR for python and nodejs.
+  Ultimately it's expected that the network will grow large enough to
+  exceed even this new maximum, so developers are planning to revamp the
+  relevant RPCs to handle this situation.
+
+- [LND #2354][] adds a new `invoicestate` field and deprecates the former
```

Comment:
I think 'removes' is better than 'deprecates'. *Deprecates* suggests to me that the field is still there but will be removed in future, but in fact PR 2354 removes the field.

Reply:
Are you sure?  I don't run lnd, so I can't quickly check, but I wrote my description based on how I guessed it worked from the patch:

```diff
diff --git a/lnrpc/rpc.proto b/lnrpc/rpc.proto
index 8794c1b4..10e53179 100644
--- a/lnrpc/rpc.proto
+++ b/lnrpc/rpc.proto
@@ -1699,7 +1699,7 @@ message Invoice {
     int64 value = 5 [json_name = "value"];
 
     /// Whether this invoice has been fulfilled
-    bool settled = 6 [json_name = "settled"];
+    bool settled = 6 [json_name = "settled", deprecated = true];
```

I don't know enough about gRPC in general or lnd in particular to know whether this removed the field or just tags it as deprecated.  However, I would consider it poor form on their part to remove a field without deprecating it first.

I rewrote the opening of this bullet so that it's consistent with either of our interpretations, but feel free to edit as you think appropriate.

Reply:
You're right and I'm wrong. This is deprecated and not removed. The `settled` field is set from the `ContractTerm.State` field here: https://github.com/lightningnetwork/lnd/pull/2354/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR3306

Reply:
I've reverted your change back to '...and deprecates the former'